eid-mw: replace libbeidpkcs11.so with opensc-pkcs11.so library

MarkRijckenberg opened the pull request:
To solve following issues:
https://github.com/ValveSoftware/steam-runtime/issues/667#issuecomment-2093093716
https://github.com/NixOS/nixpkgs/issues/298662

An issue with libpcsclite_real.so.1 is causing the build of eid-mw version 5.1.16 to fail:
https://hydra.nixos.org/build/258123099/nixlog/1/tail
See also:
https://github.com/NixOS/nixpkgs/issues/309158
Should hopefully get resolved via following merge request for pcsclite:
https://github.com/NixOS/nixpkgs/pull/308884

## Description of changes

<!--
This library change from  

 libfile="/run/current-system/sw/lib/libbeidpkcs11.so" 
to
libfile="/run/current-system/sw/lib/opensc-pkcs11.so"
is to avoid crashloop of steamwebhelper (part of steam package) due to steamwebhelper reading from /run/current-system/sw/lib/libbeidpkcs11.so 
There is a bug in /run/current-system/sw/lib/libbeidpkcs11.so causing steamwebhelper to crashloop.

/run/current-system/sw/lib/opensc-pkcs11.so does not cause a crashloop of steamwebhelper
/run/current-system/sw/lib/opensc-pkcs11.so supports the use of Belgian eid card readers.

According to https://wiki.debian.org/BeID, the Belgian-eID-specific fork of OpenSC was merged back into mainline OpenSC, so it is not necessary to use libbeidpkcs11.so any more: the more generic opensc-pkcs11.so should be sufficient.

-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)  
Successfully tested modified version of eid-nssdb script
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MarkRijckenberg commented:
An issue with libpcsclite_real.so.1 is causing the build of eid-mw version 5.1.16 to fail:

https://hydra.nixos.org/build/258123099/nixlog/1/tail

See also:

https://github.com/NixOS/nixpkgs/issues/309158

DontEatOreo commented:
Please squash 3b8854271a00cab87efa9d9d169eac047bd11ba4 into f42434a8d78044ab26c00d891703aa5069b6e0b1 and reword f42434a8d78044ab26c00d891703aa5069b6e0b1 so you can comply with the [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) and then update your PR title to be the same as commit title.

MarkRijckenberg commented:
@DontEatOreo: Thanks for your remarks. As you know, I am just a first-time contributor. I tried to squash and merge the 2 commits, but I do not think I have enough access rights to perform that action here on Github. Maybe because I am not the maintainer for the eid-mw package?

MarkRijckenberg committed:
eid-mw: replace libbeidpkcs11.so with opensc-pkcs11.so library

To solve following issues:
https://github.com/ValveSoftware/steam-runtime/issues/667#issuecomment-2093093716

https://github.com/NixOS/nixpkgs/issues/298662

MarkRijckenberg commented:
So now we have a situation where there is still a conflict between eid-mw and steam due to the libbeidpkcs11.so  library.

At the same time, the eid-mw team state:

Conflict with the OpenSC PKCS#11 module
The OpenSC module claims it supports the Belgian eID card, but it only
supports applet 1.7, not 1.8. The result is that users who have OpenSC
installed may or may not successfully authenticate, depending on whether
their browser prefers OpenSC over the eID software (or not).

To avoid this situation, we can conflict with OpenSC. Users who need
OpenSC for other things will hate us for that, but that can't be helped.

See here:
https://github.com/Fedict/eid-mw/commit/11daff9b62d49db719579c5bb39063bd796dbd89

However, I can tell you that I tested eid-nssdb with opensc-pkcs11.so library on google-chrome-stable browser and the eid card reader worked fine with that combination.....

MarkRijckenberg commented:
Sorry everyone. According to following feedback
https://github.com/ValveSoftware/steam-runtime/issues/667#issuecomment-2107095792
we cannot use opensc-pkcs11.so library as replacement for libbeidpkcs11.so, because the Beid specific fork of opensc is not maintained by anyone and because the workaround of using opensc-pkcs11.so only works for Belgian eid cards issued around 4 to 5 years ago..... So I am closing this pull request..... My eid card was issued in September 2020, so maybe it was one of the last cards to work with the opensc-pkcs11.so library....

MarkRijckenberg commented:
I have solved the crashlooping issue by performing a complete reinstall of the steam client on NixOS unstable.

I ran rm -rf ~/.local/share/Steam
Then reran steam to force a new install.
No more crashlooping of steam, even when libbeidpkcs11.so is loaded via eid-nssdb add
